### PR TITLE
Update product-grid-toggle with deferred rendering

### PR DESCRIPTION
**Pull request by ScottPolhemus**
This branch implements a proof-of-concept for deferring the liquid rendering logic for additional product grids which are hidden on the initial page load in the `product-grid-toggle` section.

- `templates/index.ajax.liquid` was created so we can request AJAX-specific variations of modules on the index page using the Section Rendering API.
- `sections/product-grid-toggle.liquid` was updated so that it only renders the first product grid when serving the index page. When requested with the section rendering API using the `view=ajax` parameter, the AJAX-specific logic is activated to only render the additional tabs. (_Note: This branch also happens to include some prior changes I made to the `product-grid-toggle` section blocks to allow for showing collections as well as curated product lists. These are not relevant to the AJAX functionality._)
- `modules/product-grid-toggle.js` was updated to request the AJAX variation of the section and inject the deferred content into the module using `DynamicSectionElement`.
- `frontend/scripts/dynamic-section-element.js` was updated to add support for an `onlySlots` option, to control which slots in the re-rendered module will be replaced. This was needed so that `product-grid-toggle` could replace the content for the additional deferred tabs without affecting the initial tab.

### CODE REVIEW

File: themes/seed-theme/sections/product-grid-toggle.liquid [hunk 1]
```diff
@@ -1,33 +1,54 @@
...
+
+      if defer_additional_tabs
+        if template.suffix != 'ajax' and forloop.index0 != 0
+          assign render_block = false
+        elsif template.suffix == 'ajax' and forloop.index0 == 0
+          assign render_block = false
+        endif
+      endif
+    %}
+    <div data-slot="product-grid-toggle-{{ forloop.index }}" class="{% if forloop.index0 > 0 %}hidden{% endif %}">
```
Comment (montalvomiguelo) [inline themes/seed-theme/sections/product-grid-toggle.liquid:33]: @ScottPolhemus we should be able to leverage [the slot element](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_templates_and_slots#adding_flexibility_with_slots) here

File: themes/seed-theme/modules/product-grid-toggle/product-grid-toggle.js [hunk 2]
```diff
@@ -1,41 +1,62 @@
...
   }
 
-  get siblingProductGridSections () {
-    return this.parentElement.querySelectorAll('[data-module="product-grid"]')
+  renderAdditionalTabs () {
+    const params = new URLSearchParams({
+      view: 'ajax',
+      section_id: this.parentElement.id.slice('shopify-section-'.length)
+    })
+    fetch(`${window.location.pathname}?${params}`)
```
Comment (montalvomiguelo) [inline themes/seed-theme/modules/product-grid-toggle/product-grid-toggle.js:50]: @ScottPolhemus is there a way we can re-use this? it handles both requesting content, replacing content for slots

https://github.com/barrel/barrel-shopify/blob/ae33314d16a13ddd4558ce2b58b650faf660532d/themes/seed-theme/frontend/scripts/dynamic-section-element.js#L11

**Comment by montalvomiguelo** [comment]
I like this pattern for retrieving and painting content from sections asynchronously. It will help improve liquid rendering times. 

I re-used loadSectionFromUrl from multiple components on [this project](https://gitlab.com/barrel/blackanddecker/-/commit/10a0a9d64495978f2a44683283f9cb1be4d2bdb2). I think we don't need to re-implement that behavior at component level.
